### PR TITLE
Convert website from Jekyll to Rails

### PR DESCRIPTION
**Pull request by jgarber623-gov**
## Checklist

I have…

- [x] served the site locally (`bundle exec rails server`) and verified that my changes behave as expected.
- [x] run the test suite (`bundle exec rake spec`) and verified that all tests pass.
- [x] summarized below my changes and noted which issues (if any) this pull request fixes or addresses.
- [x] thoroughly outlined below the steps necessary to test my changes.

## Summary of Changes

This pull request migrates the website from being a static, generated site using [Jekyll](http://jekyllrb.com) to a full-blown [Rails](http://rubyonrails.org) application. The team decided on this after discussing a number of upcoming features that would push beyond the capabilities of a static website and that would be best-served by an application and a database.

The PR adds stock Rails 5.1 application skeleton plus [RSpec](http://rspec.info), [Rubocop](http://rubocop.readthedocs.io), and [Simplecov](https://github.com/colszowka/simplecov) for better application development workflow.

## Testing

To verify the changes proposed in this pull request…

1. clone this repo,
1. `git checkout rails-app`,
1. set up development dependencies according to the modified `CONTRIBUTING.md`,
1. `bin/rails server`, and
1. `bin/rake spec` (for good measure).

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -4,8 +4,8 @@ Thanks for submitting a pull request! Below are a few things you can do to help
 
 I have…
 
-- [ ] built and served the site locally (`bundle exec rake jekyll:serve`) and verified that my changes behave as expected.
-- [ ] run the test suite (`bundle exec rake`) and verified that all tests pass.
+- [ ] run the application locally (`bundle exec rails server`) and verified that my changes behave as expected.
```
Comment (tpitale) [inline .github/PULL_REQUEST_TEMPLATE.md]: `bin/rails server` may be better (for consistency in the near future)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -12,63 +12,47 @@ There are several ways in which you can help improve this project:
...
 
-After successfully installing Bundler, run the following command from the root of the project to install the dependencies specified in the [Gemfile][gemfile]:
+With your PostgreSQL server running, issue the following command from the root of the project:
 
 ```sh
-bundle install
+./bin/setup
 ```
 
+This setup script will install the dependencies specified in the [Gemfile][gemfile], create `config/database.yml` and `config/secrets.yml`, and create and migrate the application's databases. When the setup script has finished, update the `config/database.yml` and `config/secrets.yml` configuration files to match your environment. You can run `bundle exec rake secret` to generate values for the `secret_key_base` keys in `config/secrets.yml`.
```
Comment (tpitale) [inline CONTRIBUTING.md]: Make this a bulleted list for readability?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -12,63 +12,47 @@ There are several ways in which you can help improve this project:
...
+
+Start the application by running `bundle exec rails server` and opening [http://localhost:3000](http://localhost:3000) in your Web browser of choice.
```
Comment (tpitale) [inline CONTRIBUTING.md]: Ditto `bin/rails server`

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -12,63 +12,47 @@ There are several ways in which you can help improve this project:
...
+
 ## Making Changes
 
 1. Fork and clone the project's repo.
 1. Install development dependencies as outlined above.
 1. Create a feature branch for the code changes you're looking to make: `git checkout -b your-descriptive-branch-name origin/master`.
 1. _Write some code!_
-1. Build the site and verify that your changes function as intended: `bundle exec rake jekyll:build`.
+1. Run the application and verify that your changes function as intended: `bundle exec rails server`.
+1. If your changes would benefit from testing, add the necessary tests and verify everything passes by running `bundle exec rake spec`.
```
Comment (tpitale) [inline CONTRIBUTING.md]: This can also be `bin/rake spec`.

File: app/assets/javascripts/cable.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+// Action Cable provides the framework to deal with WebSockets in Rails.
```
Comment (tpitale) [inline app/assets/javascripts/cable.js]: Are you planning to use `ActionCable`?

File: .rspec [hunk 6]
```diff
@@ -0,0 +1,3 @@
+--color
+--format progress
+--require rails_helper
```
Comment (jgarber623-gov) [inline .rspec]: Requiring `rails_helper` here so we don't need to add `require 'rails_helper'` to the top of every spec file.

File: .simplecov [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+
+formatters = [SimpleCov::Formatter::HTMLFormatter]
+
+if RSpec.configuration.files_to_run.length > 1
+  formatters << SimpleCov::Formatter::Console
+end
+
+SimpleCov.start 'rails' do
+  formatter SimpleCov::Formatter::MultiFormatter.new(formatters)
+end
```
Comment (jgarber623-gov) [inline .simplecov]: This file configures Simplecov and will generate a helpful coverage table on the console when there's more than one spec being run.

File: app/assets/javascripts/cable.js (hunk 5 above)
Comment (jgarber623-gov) [inline app/assets/javascripts/cable.js]: @tpitale 99% sure we won't. Given that, probably worth ditching:

- `app/assets/javascripts/channels`
- `app/assets/javascripts/cable.js`
- `app/channels`

…yeah?

Comment (tpitale) [inline app/assets/javascripts/cable.js]: Yeah, alllll that jonx can go.

Comment (tpitale) [inline app/assets/javascripts/cable.js]: Can always bring it back later. Not hard to restore.

**Pull request merged**